libyara/modules/pe.c

```diff
@@ -1052,7 +1052,7 @@ int pe_valid_dll_name(
     }
   }
 
-  return (l > 0 && l < n);
+  return (l > 0 && l <= n);
```

Comment:
You're correct, of course. Turns out that the use of a `strdup` style function is wrong here. But then, we didn't know about the string-at-EOF corner case before...